File: Task 1/counter_ws/src/counter_pkg/scripts/sub_count.py
#!/usr/bin/python3

import rospy
from std_msgs.msg import Int32


def listener():
    # create sub node
    rospy.init_node("Listener", anonymous= True)
    # subscribe /count_topic
    rospy.Subscriber("count_topic", Int32, callback)
    rospy.spin()  # process incoming messages until the node is shutdown


def callback(count):  # callback function to handle msg
    rospy.loginfo("count = %d", count.data)



if __name__ == "__main__":
    listener()

File: pub & sub/task1_ws/src/counter_pkg/src/sub_count.cpp
#include "ros/ros.h"
#include "std_msgs/Int32.h"

using namespace ros;
// callback function to handle message
void callback(const std_msgs::Int32::ConstPtr &msg)
{
    // print msg on console screen
    ROS_INFO("count = %d", msg->data);
}
int main(int argc, char **argv)
{
    // node
    init(argc, argv, "listener", init_options::AnonymousName);
    NodeHandle nh;

    // Subscriber
    Subscriber sub = nh.subscribe("count_topic", 1, callback);
    spin();

    return 0;
}